File: POG/anagram.c
#include<stdio.h>
void main()
{
	char str1[50],str2[50];
	printf("Enter String:\n");
	scanf("%s",&str1);
	printf("Enter String:\n");
	scanf("%s",&str2);
	int k=anagram(str1,str2);
	if(k==1)
	{
		printf("Two Strings are Anagram");
	}
	else
	{
		printf("Two Strings are Not Anagram");
	}
}
int anagram(char str1[],char str2[])
{
	int flag=0,i,j;
	char ch;
	if(strlen(str1)!=strlen(str2))
	return 0;
	flag=0;
	for(i=0;str1[i]!='\0';i++)
	{
		for(j=0;str2[j]!='\0';j++)
		{
			if(str1[i]==str2[j])
			{	flag=1;
				ch=str2[i];
				str2[i]=str2[j];
				str2[j]=ch;
				
				if(strcmp(str1,str2)==0)	return 1;
			}
		}
		if(flag==0)
		return 0;
    }  
}

File: Core Java/Anagram/src/study/Anagram.java
package study;

public class Anagram {

	public static void main(String[] args) {
		String str1=new String("abcd");
		String str2=new String("dcbaa");
		char[] arr1=str1.toCharArray();
		char[] arr2=str2.toCharArray();
		Boolean flag=false;
		char ch;
		if(str1.length()!=str2.length())
		{
			flag=false;
		}
		else {
			for(int i=0;i<arr1.length;i++)
			{
				for(int j=0;j<arr2.length;j++)
				{
					if(arr1[i]==arr2[j])
					{
						flag=true;
						ch=arr2[i];
						arr2[i]=arr2[j];
						arr2[j]=ch;
					}
				}
				if(arr1[i]==arr2[i])
				{
					flag=true;
					continue;
				}
				else
				{
					flag=false;
					break;
				}
			}
			}
			if(flag)
			{
				System.out.println("Anagram");
			}
			else
			{
				System.out.println("Not Anagram");
			}
		
	}

}
